Microsoft X-Box One Elite 2 pad + Microsoft Xbox Series S|X Controller

JGoutin opened the pull request:
Same controllers as #286 but detected differently when USB wired on a second Kodi setup.

Note: The `controller` XML section content for all theses Xbox seems to be always the same, maybe it can be used as default when a new unknown controller is detected ?

JGoutin committed:
Microsoft X-Box One Elite 2 pad + Microsoft Xbox Series S|X Controller (USB Wired)

garbear commented:
The `<configuration>` sections are different, possibly due to confusion when detecting the triggers. In order to handle so many driver quirks we do a lot of detection and correction behind the scenes. Can you verify if both configuration sections are correct?

> Note: The controller XML section content for all theses Xbox seems to be always the same, maybe it can be used as default when a new unknown controller is detected ?

I started implementing a feature called "Joystick Families": https://github.com/xbmc/peripheral.joystick/blob/Omega/peripheral.joystick/resources/joystickfamilies/joystickfamilies.xml but unfortunately never finished it.

JGoutin committed:
Uniformize axis configuration for Xbox controllers.

JGoutin commented:
I guess the `axis` entries in `configuration` are missing sometime. I tested with or without `axis` entries with all my controllers and configs, I don't find difference if present or not. I updated all config to have `axis` entries and tested again, this worked in all cases.

The auto-detection process in the GUI has some weird behavior sometime: the popup for analog axis detection can appear at any time in the process (Randomly after any button selection, not always when all buttons were selected), in that case, the profile has some missing buttons and the operation need to be restarted. This may be the reasons for the differences.

---
I also found why I have 2 profiles for the sames wired controllers, there is one for the native Linux kernel driver ("xpad") and on for the third party [xone](https://github.com/medusalix/xone) Kernel module (That adds support for the Microsoft wireless proprietary dongle, but also replace the wired driver).

This PR adds the native kernel "xpad" profiles, #286 added the "xone" driver profiles.

---

> > Note: The controller XML section content for all theses Xbox seems to be always the same, maybe it can be used as default when a new unknown controller is detected ?
> 
> I started implementing a feature called "Joystick Families": https://github.com/xbmc/peripheral.joystick/blob/Omega/peripheral.joystick/resources/joystickfamilies/joystickfamilies.xml but unfortunately never finished it.

That look good :+1:. I added the new controllers to the related XML.

Suggestion: "*" wildcard or regex support to automatically handle many controllers at once and future variants.

Example:
```xml
<joystick>Microsoft X-Box * pad</joystick>
```
Replaces:
```xml
<joystick>Microsoft X-Box 360 pad</joystick>
<joystick>Microsoft X-Box One pad</joystick>
<joystick>Microsoft X-Box One Elite pad</joystick>
<joystick>Microsoft X-Box One Elite 2 pad</joystick>
```

JGoutin committed:
Add more XBox controllers to Joystick families

JGoutin commented:
> > Note: The controller XML section content for all theses Xbox seems to be always the same, maybe it can be used as default when a new unknown controller is detected ?
> 
> I started implementing a feature called "Joystick Families": https://github.com/xbmc/peripheral.joystick/blob/Omega/peripheral.joystick/resources/joystickfamilies/joystickfamilies.xml but unfortunately never finished it.

That look good :+1:. I added the new controllers to the related XML.

Suggestion: "*" wildcard or regex support to automatically handle many controllers at once and future variants.

Example:
```xml
<joystick>Microsoft X-Box * pad</joystick>
```
Replaces:
```xml
<joystick>Microsoft X-Box 360 pad</joystick>
<joystick>Microsoft X-Box One pad</joystick>
<joystick>Microsoft X-Box One Elite pad</joystick>
<joystick>Microsoft X-Box One Elite 2 pad</joystick>
```

garbear commented:
Wildcard or regex is a good idea - or maybe even fuzzy string matching. Unfortunately joystickfamilies.xml isn't hooked up in the C++, I just created the XML file based on the joystick data I removed from Kodi when the new controller work was added in v17. I didn't wanna lose the data but never finished the code. It wouldn't be too hard to add though.

> The auto-detection process in the GUI has some weird behavior sometime

The weird GUI behavior is because joystick behavior is even weirder across all platforms we support. Something I affectionately refer to as "joystick driver fuckery". My greatest feat was encapsulating all the ugliness behind clean APIs, so at least the code is pristine, and the GUI came out only a "little" weird.
